R/sass.R

```diff
@@ -142,6 +150,42 @@ sass <- function(input = NULL, options = sass_options(), output = NULL,
   css
 }
 
+#' An intelligent (temporary) output file
+#'
+#' Provide the return value of this function to [sass()]'s `output` argument for
+#' a temporary `output` file that is `cache` and `options` aware (see details).
+#'
+#' When `cache` is enabled, the file path is `file.path(tempdir(),
+#' paste0(pattern, cache_key), paste0(basename, fileext))`. This ensures that
+#' when [sass()] has a cache hit, it won't needlessly generate redundant
+#' file(s).
+#'
+#' @param basename a non-empty character vector giving the outfile name (without
+#'   the extension).
+#' @param pattern a non-empty character vector giving the initial part of the
+#'   directory name.
+#' @param fileext the output file extension. The default is `".min.css"` for
+#'   compressed and compact output styles; otherwise, its `".css"`.
+#' @export
+#' @examples
+#' sass("body {color: red}", output = output_file())
+output_file <- function(basename = "sass", pattern = basename, fileext = NULL) {
```

Comment:
Instead of `pattern`, I think it would be clearer to use `dirname`. I know that it isn't exactly the directory name (since a suffix gets added), but `pattern` is vague and could lead the user to think it refers to something else.

Reply:
Ok, yea. Just FYI, I went with `pattern` since it's basically equivalent to `tempfile()`'s `pattern`